Snowflake.Data.Tests/SFDataConverterTest.cs

```diff
@@ -97,5 +96,96 @@ private void internalTestConvertDate(DateTime dtExpected, DateTime testValue)
+            Int64 expected = Convert.ToInt64(s);
+            Assert.AreEqual(expected, actual);
+        }
+
+        [Test]
+        [TestCase("2147483647")]
+        [TestCase("-2147483648")]
+        [TestCase("-1")]
+        [TestCase("0")]
+        public void TestConvertToInt32(string s)
+        {
+            Int32 actual = (Int32)SFDataConverter.ConvertToCSharpVal(s, SFDataType.FIXED, typeof(Int32));
+            Int32 expected = Convert.ToInt32(s);
+            Assert.AreEqual(expected, actual);
+        }
+
+        [Test]
+        [TestCase("32767")]
+        [TestCase("-32768")]
+        [TestCase("-1")]
+        [TestCase("0")]
+        public void TestConvertToInt16(string s)
+        {
+            Int16 actual = (Int16)SFDataConverter.ConvertToCSharpVal(s, SFDataType.FIXED, typeof(Int16));
+            Int16 expected = Convert.ToInt16(s);
+            Assert.AreEqual(expected, actual);
+        }
+
+        [Test]
+        [TestCase("255")]
+        [TestCase("0")]
+        public void TestConvertToByte(string s)
+        {
+            byte actual = (byte)SFDataConverter.ConvertToCSharpVal(s, SFDataType.FIXED, typeof(byte));
+            byte expected = Convert.ToByte(s);
+            Assert.AreEqual(expected, actual);
+        }
+
+        [Test]
+        [TestCase("256")]
+        [TestCase("-1")]
+        public void TestOverflowByte(string s)
+        {
+            Assert.Throws<OverflowException>(() => SFDataConverter.ConvertToCSharpVal(s, SFDataType.FIXED, typeof(byte)));
+        }
+
+        [Test]
+        [TestCase("32768")]
+        [TestCase("-32769")]
+        public void TestOverflowInt16(string s)
+        {
+            Assert.Throws<OverflowException>(() => SFDataConverter.ConvertToCSharpVal(s, SFDataType.FIXED, typeof(Int16)));
+        }
+
+        [Test]
+        [TestCase("2147483648")]
+        [TestCase("-2147483649")]
+        public void TestOverflowInt32(string s)
+        {
+            Assert.Throws<OverflowException>(() => SFDataConverter.ConvertToCSharpVal(s, SFDataType.FIXED, typeof(Int32)));
+        }
+
+        [Test]
+        [TestCase("9223372036854775808")]
+        [TestCase("-9223372036854775809")]
+        public void TestOverflowInt64(string s)
+        {
+            Assert.Throws<OverflowException>(() => SFDataConverter.ConvertToCSharpVal(s, SFDataType.FIXED, typeof(Int64)));
+        }
+
+        [Test]
+        [TestCase("9223372036854775807.9223372036854775807")]
+        [TestCase("-9223372036854775807.1234567890")]
+        [TestCase("-1.300")]
+        [TestCase("999999999999999999.000000000000100000000000")]
+        [TestCase("4294967295.4294967296")]
+        public void TestConvertToDecimal(string s)
+        {
+            decimal actual = (decimal)SFDataConverter.ConvertToCSharpVal(s, SFDataType.FIXED, typeof(decimal));
+            decimal expected = Convert.ToDecimal(s, CultureInfo.InvariantCulture);
+
+            Assert.AreEqual(expected, actual);
+        }

```

Comment:
Can we had a few extra tests unsupported conversions and a couple of special cases?
```     }

        [Test]
        [TestCase("9223372036854775807.9223372036854775807")]
        [TestCase("-9223372036854775807.1234567890")]
        [TestCase("-1.300")]
        [TestCase("999999999999999999.000000000000100000000000")]
        [TestCase("4294967295.4294967296")]
        [TestCase("1.5e-36")]
        [TestCase("1.5e+38")]
        //[TestCase("inf")] -- TODO - Not supported yet
        //[TestCase("-inf")] -- TODO - Not supported yet
        [TestCase("NaN")]
        public void TestConvertToFloat(string s)
        {
            double actualDouble = (double)SFDataConverter.ConvertToCSharpVal(s, SFDataType.FIXED, typeof(double));
            double expectedDoulbe = Convert.ToDouble(s, CultureInfo.InvariantCulture);

            Assert.AreEqual(actualDouble, expectedDoulbe);

            float actualFloat = (float)SFDataConverter.ConvertToCSharpVal(s, SFDataType.FIXED, typeof(float));
            float expectedFloat = Convert.ToSingle(s, CultureInfo.InvariantCulture);

            Assert.AreEqual(expectedFloat, actualFloat);
        }

        [Test]
        [TestCase("thisIsNotAValidValue")]
        [TestCase("-1.300")]
        [TestCase("425.426")]
        [TestCase("1.5e-36")]
        [TestCase("1.5e+38")]
        [TestCase("inf")]
        [TestCase("-inf")]
        [TestCase("NaN")]
        public void TestInvalidConversionInvalidInt(string s)
        {
            Assert.Throws<FormatException>(() => SFDataConverter.ConvertToCSharpVal(s, SFDataType.FIXED, typeof(Int32)));
            Assert.Throws<FormatException>(() => SFDataConverter.ConvertToCSharpVal(s, SFDataType.FIXED, typeof(Int64)));
            Assert.Throws<FormatException>(() => SFDataConverter.ConvertToCSharpVal(s, SFDataType.FIXED, typeof(Int16)));
            Assert.Throws<FormatException>(() => SFDataConverter.ConvertToCSharpVal(s, SFDataType.FIXED, typeof(byte)));
        }

        [Test]
        [TestCase("thisIsNotAValidValue")]
        public void TestInvalidConversionInvalidFloat(string s)
        {
            Assert.Throws<FormatException>(() => SFDataConverter.ConvertToCSharpVal(s, SFDataType.FIXED, typeof(float)));
            Assert.Throws<FormatException>(() => SFDataConverter.ConvertToCSharpVal(s, SFDataType.FIXED, typeof(double)));
        }

        [Test]
        [TestCase("thisIsNotAValidValue")]
        [TestCase("1.5e-36")]
        [TestCase("1.5e+38")]
        [TestCase("inf")]
        [TestCase("-inf")]
        [TestCase("NaN")]
        public void TestInvalidConversionInvalidDecimal(string s)
        {
            Assert.Throws<FormatException>(() => SFDataConverter.ConvertToCSharpVal(s, SFDataType.FIXED, typeof(decimal)));
        }

```

Reply:
Added these tests to the PR